app/src/main/AndroidManifest.xml

```diff
@@ -8,6 +8,12 @@
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
     <uses-permission android:name="android.permission.WAKE_LOCK" />
 
+    <permission
+        android:name="net.christianbeier.droidvnc_ng.VNC_IPC_CONTROL"
+        android:label="@string/permlab_vncIpcControl"
+        android:description="@string/permdesc_vncIpcControl"
+        android:protectionLevel="dangerous" />
```

Comment:
Who would give this permission how? Can you document the UX flow please?

Reply:
https://developer.android.com/training/permissions/requesting
1. Declare the permission in the app's manifest file
2.

Reply:
https://developer.android.com/training/permissions/requesting
1. Declare the permission in the app's manifest file
`<uses-permission android:name="net.christianbeier.droidvnc_ng.VNC_IPC_CONTROL" />`
2.  In your initialization logic, check the permission

```
private void CheckPermissions()
{
	var permission = ContextCompat.CheckSelfPermission(this, "net.christianbeier.droidvnc_ng.VNC_IPC_CONTROL");
	if (permission == Android.Content.PM.Permission.Granted)
	{
		// You can use the API that requires the permission.
	}
	else
	{
		// You can directly ask for the permission.
		RequestPermissions(new String[] { "net.christianbeier.droidvnc_ng.VNC_IPC_CONTROL" }, 777);
	}
}
```

3. After the user responds to the system permissions dialog, the system then invokes your app's implementation of `[onRequestPermissionsResult()]`. The system passes in the user response to the permission dialog, as well as the request code that you defined, as shown in the following code snippet:
		
```
public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
{
	switch (requestCode)
	{
		case 777:
			// If request is cancelled, the result arrays are empty.
			if (grantResults.Length > 0 && grantResults[0] == Permission.Granted)
			{
				// Permission is granted. Continue the action or workflow
				// in your app.
			}
			else
			{
				// Explain to the user that the feature is unavailable because
				// the features requires a permission that the user has denied.
				// At the same time, respect the user's decision. Don't link to
				// system settings in an effort to convince the user to change
				// their decision.
			}
			return;
	}
	// Other 'case' lines to check for other
	// permissions this app might request.


	base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
}

```

Reply:
Any application needing an access to the `droidVNC-NG` resources has to follow workflow for requesting permissions.
1. Declare the VNC permission in the app's manifest file
`<uses-permission android:name="net.christianbeier.droidvnc_ng.VNC_IPC_CONTROL" />`
2. The next steps depend on the usage model of the device and the  application. In some cases, the check is performed during application startup, in others, when the user starts to interact with the VNC feature. In any case, it is necessary to check whether the `droidVNC-NG` is installed on this device. If not so, inform the user about the absence of the application; otherwise continue to the next step.
3. Check whether the user has already granted the runtime permission to control the VNC. If so, your app can control the VNC. If not, continue to the next step. 
4. Request the VNC permission that your app requires in order to access the VNC feature. The system displays a runtime permission prompt.
5. Check the user's response, whether they chose to grant or deny the runtime permission.
6. If the user granted the permission to your app, you can access the VNC feature.